core/src/scene/importer.cpp

```diff
@@ -162,43 +162,37 @@ std::vector<Url> Importer::getResolvedImportUrls(const Node& sceneNode, const Ur
                     }
                 }
             }
+            Node(sceneNode).remove("import");
         }
     }
 
     return sceneUrls;
 }
 
-void Importer::importScenesRecursive(Node& root, const Url& sceneUrl, std::vector<Url>& sceneStack) {
+void Importer::importScenesRecursive(Node& root, const Url& sceneUrl, std::unordered_set<Url>& imported) {
 
     LOGD("Starting importing Scene: %s", sceneUrl.string().c_str());
 
-    for (const auto& s : sceneStack) {
-        if (sceneUrl == s) {
-            LOGE("%s will cause a cyclic import. Stopping this scene from being imported",
-                    sceneUrl.string().c_str());
-            return;
-        }
-    }
-
-    sceneStack.push_back(sceneUrl);
+    // Insert self to handle self-imports cycles
+    imported.insert(sceneUrl);
 
-    auto sceneNode = m_importedScenes[sceneUrl];
+    auto& entry = m_importedScenes[sceneUrl];
+    auto& sceneNode = entry.first;
+    auto& imports = entry.second;
```

Comment:
nit: `imports` and `imported` can easily be messed up in future refactor. I would maybe rename `imports` to `currentSceneImports` or something on those line.